src/image-alt-text/opportunityHandler.js

```diff
@@ -11,27 +11,39 @@
  */
 
 import { isNonEmptyArray } from '@adobe/spacecat-shared-utils';
-import { Audit } from '@adobe/spacecat-shared-data-access';
+import { Audit as AuditModel, Suggestion as SuggestionModel } from '@adobe/spacecat-shared-data-access';
+
+const getImageSuggestionIdentifier = (suggestion) => `${suggestion.pageUrl}/${suggestion.src}`;
 
 /**
  * Synchronizes existing suggestions with new data
  * by removing existing suggestions and adding new ones.
  *
  * @param {Object} params - The parameters for the sync operation.
  * @param {Object} params.opportunity - The opportunity object to synchronize suggestions for.
- * @param {Array} params.newSuggestions - Array of new data objects to sync.
+ * @param {Array} params.newSuggestionDTOs - Array of new data objects (not models) to sync.
  * @param {Object} params.log - Logger object for error reporting.
  * @returns {Promise<void>} - Resolves when the synchronization is complete.
  */
-export async function syncAltTextSuggestions({ opportunity, newSuggestions, log }) {
+export async function syncAltTextSuggestions({ opportunity, newSuggestionDTOs, log }) {
   const existingSuggestions = await opportunity.getSuggestions();
 
-  // Remove existing suggestions
-  await Promise.all(existingSuggestions.map((suggestion) => suggestion.remove()));
+  // eslint-disable-next-line max-len
+  const ignoredSuggestions = existingSuggestions.filter((s) => s.getStatus() === SuggestionModel.STATUSES.SKIPPED);
```

Comment:
we should not simply get all suggestions and then programmatically filter... instead use the relevant accessors:
```js
await opportunity.getSuggestionsByStatus('NEW');
```

Reply:
https://github.com/adobe/spacecat-audit-worker/pull/683/commits/46c39f419c2da26ca5a970cf3b8b88a6109b30b4